### PR TITLE
fix: update selectors for new ucc ui

### PR DESCRIPTION
**Pull request by harshpatel-crest**
Jira: https://jira.splunk.com/browse/ADDON-36754

Changes:

- Update selectors for new UCC UI

### CODE REVIEW

File: pytest_splunk_addon_ui_smartx/components/controls/single_select.py [hunk 1]
```diff
@@ -130,16 +190,36 @@ def list_of_values(self):
...
+            self.elements.update({
+                "values":Selector(select=popoverid + ' [data-test="option"]')
+            })
+        else:
+            if self.searchable:
+                popoverid = '#' + self.dropdown.get_attribute("data-test-popover-id")
+                self.elements.update({
+                        "input": Selector(select=popoverid + ' [data-test="textbox"]')
+                    })
+            popoverid = '#' + self.dropdown.get_attribute("data-test-popover-id")
```
Comment (pdudhaiya-crest) [inline pytest_splunk_addon_ui_smartx/components/controls/single_select.py:209]: this statement is getting repeated 2 times in the else block.. Can we move it outside ?

File: pytest_splunk_addon_ui_smartx/components/controls/single_select.py [hunk 2]
```diff
@@ -130,16 +190,36 @@ def list_of_values(self):
         self.dropdown.click()
         first_element = None
         list_of_values = []
+
+        if self.allow_new_values:
+            if self.seachable:
```
Comment (dhruvanshus-crest) [inline pytest_splunk_addon_ui_smartx/components/controls/single_select.py:195]: Typo error: self.searchable

**Review by pdudhaiya-crest — APPROVED** [review]

File: pytest_splunk_addon_ui_smartx/components/entity.py [hunk 3]
```diff
@@ -19,24 +19,28 @@ class Entity(BaseComponent):
     The instance of the class holds all the controls in the entity and provides the generic interaction that can be done with the entity
     """
     
-    def __init__(self, browser, container, add_btn=None):
+    def __init__(self, browser, container, add_btn=None, is_single_page=False):
         """
             :param browser: The selenium webdriver
             :param container: Container in which the Entity is located. Of type dictionary: {"by":..., "select":...}
             :param add_btn: The locator of add_button with which the entity will be opened
+            :param is_single_page:
```
Comment (pdudhaiya-crest) [inline pytest_splunk_addon_ui_smartx/components/entity.py:27]: The description is missing here.

File: pytest_splunk_addon_ui_smartx/components/controls/single_select.py [hunk 4]
```diff
@@ -14,39 +14,48 @@ class SingleSelect(BaseControl):
...
         """
-        if open_dropdown:
-            self.wait_to_be_clickable("dropdown")
-            self.dropdown.click()
-        
+        self.wait_to_be_clickable("dropdown")
+        self.dropdown.click()
+
+        if self.allow_new_values:
+            if self.get_value():
```
[the remark this answers is not in the document]
Comment (harshpatel-crest) [reply pytest_splunk_addon_ui_smartx/components/controls/single_select.py:47]: The cancel button will be hidden when empty for `ComboBox`. So yes the condition is required.

**Review by dhruvanshus-crest — APPROVED** [review]

**Comment by harshpatel-crest** [comment]
Closing this as a duplicate to: https://github.com/splunk/addon-factory-smartx-ui-test-library/pull/135

**Pull request closed without merging**